Revamp stater code (#64)
codecrafters-io/docs: docs/contributors/adding-language-support/starter-code.mdx

---
title: "Starter Code"
description: "Learn about how to implement starter code for a language"
---

Starter code is the code that we provide to users when they start a challenge. It's the code that they'll be building on top of.

[Here's an example](https://github.com/codecrafters-io/build-your-own-redis/tree/main/compiled_starters/python).

> ## 1. Copy from Language Template

Reviewer: Run `course-sdk add-language`

Use [course-sdk](https://github.com/codecrafters-io/course-sdk) to copy the starter code skeleton and Dockerfile from the [language-templates](https://github.com/codecrafters-io/language-templates) repository.

<Steps>
<Step title="Clone build-your-own-<challenge>">
  For example, if you’re adding support for the Kafka challenge, clone the [build-your-own-kafka](https://github.com/codecrafters-io/build-your-own-kafka) repository.

</Step>

<Step title="Run course-sdk add-language <language>">

> a. Verify that the language you want to add is available in the [languages](https://github.com/codecrafters-io/language-templates/tree/main/languages) folder in the language-templates repository.

Reviewer: Let's remove this - `course-sdk` will automatically print an error message if the language isn't supported

b. Install [course-sdk](https://github.com/codecrafters-io/course-sdk) if it’s not already installed.

> c. In the cloned challenge repository, copy the starter code skeleton and Dockerfile with this command:

Reviewer: ```suggestion
c. In the cloned challenge repository, run the following command:
```

```bash
course-sdk add-language <language>
```

</Step>
</Steps>

> ## 2. Implement the starter code

Reviewer: ```suggestion
## 2. Edit starter code
```
